### ISSUE TITLE
Test assemblies no longer resolve properly in "no AppDomain" mode

### ISSUE THREAD

**Issue by ericeil** [xunit/xunit#592]
See my comment on 4089905f0e7ba13160f6048b32b3a3988eb203dc.  The change to call a different overload of `Assembly.Load` seems to have broken assembly resolution in the no-AppDomain case; the previous code would use the CodeBase set in the AssemblyName to find the assembly, but the new code uses the CodeBase of the AppDomain, and so can't find the test assembly unless it's in just the right location.

**Comment by bradwilson** [comment]
Do you have a repro project?

**Comment by ericeil** [comment]
The repro is a little complex - I'm working on a performance testing tool that does test discovery in a separate process from the test runner, and that separate process is now failing to find the test assembly when I move to RC1.  It's possible that I'm doing something wrong. :)  I'll dig in a little more....

**Comment by bradwilson** [comment]
The salient issue is that "no app domain" mode cannot set the base path for assembly resolution. Note that this is only marginally related to the commit you commented on, since that's used when linked against the `dotnet` family. The `dotnet` family never uses app domains, but you can link against the `desktop` family and still turn off app domains.

**Comment by ericeil** [comment]
Right, but it used to be that `desktop` would use Assembly.Load(AssemblyName), and the AssemblyName had the full path to the assembly in it.  Now we're using Assembly.Load(string), so we lose the path to the assembly.

What I don't understand is why it works when I use xunit.console.exe to run the tests, but it doesn't work when I use my custom test discoverer thing.

**Comment by bradwilson** [comment]
Do you do this? https://github.com/xunit/xunit/blob/789ac3bbcaaf5fd860bbeb8dec70e81e39eb373d/src/xunit.console/Program.cs#L238

**Comment by ericeil** [comment]
Turns out I needed to do two things:

1.  I had a copy of xunit.execution.desktop.dll in the directory with my "runner."  That was being loaded instead of the one in the test's directory.  From there, we seemed to get lost somehow.  I reworked my build so that file doesn't get placed there any more.
2.  I also needed the AssemblyHelper trick you mentioned.

Neither of these were necessary in Beta4; the way Assembly.Load was getting called before, we were just able to find the test assembly.

Thanks for the help; I'm unblocked now.
